### ISSUE TITLE
Dropout layer position not consistent with the original implementation

### ISSUE THREAD

**Issue by yuanmao** [lucidrains/perceiver-pytorch#57]
Hi,

Thanks for this incredibly clean implementation. I had a question regarding the positioning of the dropout layer.

It appeared twice in [(the original implementation)](https://github.com/deepmind/deepmind-research/blob/826ff89f21e5143dc68ff7cb33f01cc6e237844d/perceiver/perceiver.py):
(1) in Attention, where it occurred right after softmax of attention scores
(2) in FeedForward, where it occurred at the end 

It seems in your implementation the dropout layer was put in a slightly different position. Was it on purpose or it should not affect the result much?

Thanks!

**Comment by lucidrains** [comment]
@yuanmao hi, i checked and you are right, so i've corrected it in 0.8.0

usually in transformers, people place dropouts in 4 places - right after attention, right after activations of feedforward, or at the end of both attention and feedforward blocks. i typically place them in the former two, so i don't even remember why i moved one to the end of the attention block.

so, if you read the follow up paper, Perceiver IO, they actually removed dropout altogether. dropouts are sort of losing popularity in transformers. in my experience, they have a small effect in countering overfitting, but only in non-overparameterized settings
